web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx

```diff
@@ -41,9 +42,31 @@ import { ResourceActionButton } from './ResourceActionButton';
 import SearchPanel from './SearchPanel';
 
 export function UnifiedResources() {
-  const teleCtx = useTeleport();
   const { clusterId, isLeafCluster } = useStickyClusterId();
   const enabled = localStorage.areUnifiedResourcesEnabled();
+
+  if (!enabled) {
+    history.replace(cfg.getNodesRoute(clusterId));
+  }
+
+  return (
+    <Wrapper
+      key={clusterId} // when the current cluster changes, remount the component
+      clusterId={clusterId}
+      isLeafCluster={isLeafCluster}
+    />
+  );
+}
+
+function Wrapper({
```

Comment:
Hmm, technically, in this case, the `UnifiedResources` function is a wrapper. Consider renaming this one.

Reply:
Yeah, I felt that this name was wrong, but I didn't have a better one. Renamed to `ClusterResources`.